frontend/ui/widget/filechooser.lua

```diff
@@ -75,6 +75,141 @@ local FileChooser = Menu:extend{
     },
     path_items = nil, -- hash, store last browsed location (item index) for each path
     goto_letter = true,
+    collates = {
+        strcoll = {
+            text = _("name"),
+            menu_order = 10,
+            can_collate_mixed = true,
+            init_sort_func = function(cache)
+                return function(a, b)
+                    return ffiUtil.strcoll(a.text, b.text)
+                end, cache
+            end,
+        },
+        natural = {
+            text = _("name (natural sorting)"),
+            menu_order = 20,
+            can_collate_mixed = true,
+            init_sort_func = function(cache)
+                local natsort
+                natsort, cache = sort.natsort_cmp(cache)
+                return function(a, b)
+                    return natsort(a.text, b.text)
+                end, cache
+            end
+        },
+        access = {
+            text = _("last read date"),
+            menu_order = 30,
+            can_collate_mixed = true,
+            init_sort_func = function(cache)
+                return function(a, b)
+                    return a.attr.access > b.attr.access
+                end, cache
+            end,
+            mandatory_func = function(item)
+                return datetime.secondsToDateTime(item.attr.access)
+            end,
+        },
+        date = {
+            text = _("date modified"),
+            menu_order = 40,
+            can_collate_mixed = true,
+            init_sort_func = function(cache)
+                return function(a, b)
+                    return a.attr.modification > b.attr.modification
+                end, cache
+            end,
+            mandatory_func = function(item)
+                return datetime.secondsToDateTime(item.attr.modification)
+            end,
+        },
+        size = {
+            text = _("size"),
+            menu_order = 50,
+            can_collate_mixed = false,
+            init_sort_func = function(cache)
+                return function(a, b)
+                    return a.attr.size < b.attr.size
+                end, cache
+            end,
+        },
+        type = {
+            text = _("type"),
+            menu_order = 60,
+            can_collate_mixed = false,
+            init_sort_func = function(cache)
+                return function(a, b)
+                    if (a.suffix or b.suffix) and a.suffix ~= b.suffix then
+                        return ffiUtil.strcoll(a.suffix, b.suffix)
+                    end
+                    return ffiUtil.strcoll(a.text, b.text)
+                end, cache
+            end,
+            item_func = function(item)
+                item.suffix = util.getFileNameSuffix(item.text)
+                return item
```

Comment:
`return` can be dropped (`item` is a table, passed by reference, is changed within a function).